### ISSUE TITLE
[BUG] Cross exchange not executing taker orders due to illegal characters

### ISSUE THREAD

**Issue by PtrckM** [CoinAlpha/hummingbot#2226]
**Describe the bug**
after filled order will give error `WARNING - APIError(code=-1100): Illegal characters found in parameter 'price'; legal range is '^([0-9]{1,20})(\.[0-9]{1,20})?$'.`

**Steps To Reproduce**
create xemm strategy
maker_market: kucoin / bittrex
taker_market: binance

**Screenshots**
![image](https://user-images.githubusercontent.com/54516858/90400089-822b9300-e0ce-11ea-8e9b-cac2368df0b3.png)

```
2020-08-17 12:25:25,849 - 1 - hummingbot.market.binance.binance_market - NETWORK - Error submitting SELL LIMIT order to Binance for 0.02565000 ETHUSDC NaN.
Traceback (most recent call last):
  File "hummingbot/market/binance/binance_market.pyx", line 884, in hummingbot.market.binance.binance_market.BinanceMarket.create_order
    order_result = await self.query_api(self._binance_client.create_order, **api_params)
  File "hummingbot/market/binance/binance_market.pyx", line 254, in query_api
    async with self._throttler.weighted_task(request_weight=request_weight):
  File "hummingbot/market/binance/binance_market.pyx", line 266, in hummingbot.market.binance.binance_market.BinanceMarket.query_api
    raise ex
  File "hummingbot/market/binance/binance_market.pyx", line 256, in hummingbot.market.binance.binance_market.BinanceMarket.query_api
    return await self._async_scheduler.call_async(partial(func, *args, **kwargs),
  File "/home/hummingbot/hummingbot/core/utils/async_call_scheduler.py", line 128, in call_async
    return await self.schedule_async_call(coro, timeout_seconds, app_warning_msg=app_warning_msg)
  File "/home/hummingbot/hummingbot/core/utils/async_call_scheduler.py", line 117, in schedule_async_call
    return await fut
  File "/home/hummingbot/hummingbot/core/utils/async_call_scheduler.py", line 80, in _coro_scheduler
    fut.set_result(await coro)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/binance/client.py", line 1149, in create_order
    return self._post('order', True, data=params)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/binance/client.py", line 210, in _post
    return self._request_api('post', path, signed, version, **kwargs)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/binance/client.py", line 181, in _request_api
    return self._request(method, uri, signed, **kwargs)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/binance/client.py", line 176, in _request
    return self._handle_response(response)
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/binance/client.py", line 200, in _handle_response
    raise BinanceAPIException(response)
binance.exceptions.BinanceAPIException: APIError(code=-1100): Illegal characters found in parameter 'price'; legal range is '^([0-9]{1,20})(\.[0-9]{1,20})?$'.

```

**Release version**
- 0.30.0

**Attachments**
- ETH-USDC (kucoin)
[logs_kuc_bin_xemm-eth_usdc.log](https://github.com/CoinAlpha/hummingbot/files/5084367/logs_kuc_bin_xemm-eth_usdc.log)

- LINK-ETH (bittrex)
[bittrex-link_eth.txt](https://github.com/CoinAlpha/hummingbot/files/5084386/bittrex-link_eth.txt)

**Comment by supervik** [comment]
Same issue in 0.30.
Xemm passes price as "Nan" to a taker exchange after maker order is filled. It surely connected with removed MARKET orders in 0.30 version. 
In _cross_exchange_market_making.pyx_ this calculates the best price:
`taker_top = taker_market.c_get_price(taker_trading_pair, False)` 
but `taker_top` is never used after the calculation then.
Please fix.

**Comment by fengkiej** [comment]
Also found same issue with Binance market due to MARKET is deprecated in 0.30.0, but I'm not sure how the best way to ensure the order can closely simulate MARKET order using LIMIT/LIMIT_MAKER. Tried to use LIMIT with `get_price_for_volume`, but some orders ends up far from the mid price.

**Comment by dennisocana** [comment]
Confirmed that the strategy cannot execute trades on the taker market when a maker order is filled. This happens on every single attempt as a result, the bot only trades on the maker side.

[logs_test_xemm.log](https://github.com/CoinAlpha/hummingbot/files/5259489/logs_test_xemm.log)

![image](https://user-images.githubusercontent.com/50150287/93849992-cae2f700-fcdf-11ea-96b0-fdaca0813eee.png)

**Comment by mifeng** [comment]
I think we should be using LIMIT TAKER orders here

**Comment by PtrckM** [comment]
fix on **dev-0.32.0**
